grammar.md

```diff
@@ -25,9 +25,15 @@ statement : (funccall SEMICOLON)
             | ifstatement
             | while_statement
             | (BREAK SEMICOLON)
+            | for_statement
+            | switch_statement
 
 while_statement : WHILE LPAREN expression RPAREN statement
 
+for_statement : FOR LPAREN expression (COMA expression)* SEMICOLON expression SEMICOLON expression (COMA expression)*
+RPAREN statement
+
```

Comment:
can a switch have 0 cases?
does a switch with 0 cases and no default compile correctly?